website/docs/getting-started/setup/deployment-architecture.md

```diff
@@ -1,52 +1,109 @@
 ---
 description: The page provides information about the deployment architecture of Appsmith.
 ---
+
 # Deployment Architecture
 
-Appsmith can be accessed through the Cloud by signing up at [app.appsmith.com](https://app.appsmith.com/user/sign-up) or deployed on-premise using the setup instructions provided in the [Self Hosting](/getting-started/setup) guide. 
+Appsmith can be accessed through the Cloud by signing up at [app.appsmith.com](https://app.appsmith.com/user/sign-up) or deployed on-premise using the setup instructions provided in the [Self Hosting](/getting-started/setup/installation-guides) guides.
 
-The Appsmith Cloud version runs on geographically distanced [Kubernetes](https://kubernetes.io/docs/concepts/overview/) clusters via AWS. This deployment strategy allows the Appsmith Cloud platform to scale and avoid downtime during outages, as well as conduct rolling updates with minimal disruptions to service. The self-hosted Appsmith runs as a single Docker container, housing all five processes in one image via Supervisord.
+In the cloud, Appsmith runs on distributed [Kubernetes](https://kubernetes.io/docs/concepts/overview/) clusters via AWS, offering scalability and high availability with minimal downtime during updates. For self-hosted environments, Appsmith runs in a single Docker container for simplified deployment.
 
-This page provides information about the self-hosted deployment architecture of Appsmith.
+This page provides an overview of the deployment architecture for Appsmith, focusing on its key components and their interactions in a Kubernetes-based environment.
 
 ## Components
 
-Appsmith is deployed as a single Docker container with a single volume for storing persistent data. A single Docker container runs the following processes to deploy and get started with Appsmith:
+Appsmith can be deployed as a single Docker container with a single volume for storing persistent data. For **production environments**, Appsmith recommends deploying Appsmith on a Kubernetes platform. The Kubernetes-based deployment ensures that all services are containerized and orchestrated efficiently to handle scaling and fault tolerance. 
+
+The diagram below illustrates the key components, their interactions, and data flow when Appsmith is deployed on a Kubernetes platform:
+
+<ZoomImage src="/img/Appsmith_Deployment_Architecture.png" alt="Appsmith Deployment Architecture" caption="Appsmith Deployment Architecture" />
+
+### Appsmith Server 
+The **Appsmith Server** is a cloud service managed by Appsmith, and acts as the control layer for managing licenses and template configurations.
+
+### Customer Data (Datasources)
```

Comment:
@sharat87 - The customer data is customer-facing as this resides in their network, and IMO is not a part of Appsmith's deployment architecture. Could you please elaborated on why should this be included?